lib/simplekiq/polling_worker.rb

```
@@ -0,0 +1,58 @@
+class PollingWorker
+  include Simplekiq::Worker
+
+  def perform(params)
+    @params = params
+    return if expired?
+
+    if job_info.status == :pending
+      retry_job!
+    else
+      send_callback
+    end
+  end
+
+  private
+
+  def job_id
+    @params[:job_id]
+  end
+
+  def job_info
+    @job_info ||= MonitoredJobStatus.new(job_id)
+  end
+
+  def job_instance
+    job_class = @params[:job_class]
+    Object.const_get(job_class.camelize).new if job_class.present?
+  rescue NameError => _e
+    # Job class does not exist
```

Dont wanna log or do something here? I guess you will get exceptions on undefined method nil?

---

Since this is just a gem, I cannot make assumptions about logging.

---

This eats the error is this intentional? how an invalid or missing job class affect things downstream?

---

It is intentional. This worker is scheduled automatically. So this exception will only be thrown if someone called it incorrectly. And that will cause this worker to retry for a while.

Since it is an edge case anyways, I will remove the error handling. We can always add it back in if we see this being a problem.